sandy/sections/mf7.py

```diff
@@ -0,0 +1,448 @@
+    >>> sec = sandy.sections.mf7._read_elastic_scattering(tls, 10, 2)
+    >>> print(_write_elastic_scattering(sec))
+     110.000000 9.992800-1          2          0          0          0  10 7  2    1
+     80.3178400 0.00000000          0          0          1          9  10 7  2    2
+              9          2                                              10 7  2    3
+     115.000000 14.7037200 188.150000 19.1224000 208.150000 20.3789200  10 7  2    4
+     228.150000 21.6526100 233.150000 21.9735500 248.150000 22.9420500  10 7  2    5
+     253.150000 23.2667100 268.150000 24.2459100 273.150000 24.5739800  10 7  2    6
+    """
+    if sec['LTHR'] == 1:
+        lines = sandy.write_cont(sec["ZA"],
+                                 sec["AWR"],
+                                 sec['LTHR'],
+                                 0,
+                                 0,
+                                 0)
+        diff = list(sec['T'].keys())[0]
+        for T, T_info in sec['T'].items():
+            if T == diff:
+                lines += sandy.write_tab1(T,
+                                          0,
+                                          T_info['LT'],
+                                          0,
+                                          T_info['NBT'],
+                                          T_info['INT'],
+                                          sec['EINT'],
+                                          T_info['S'])
+            else:
+                lines += sandy.write_list(T,
+                                          0,
+                                          T_info['LI'],
+                                          0,
+                                          0,
+                                          T_info['S'])
+    elif sec['LTHR'] == 2:
+        lines = sandy.write_cont(sec["ZA"], sec["AWR"], sec['LTHR'], 0, 0, 0)
+        lines += sandy.write_tab1(sec['SB'],
+                                  0,
+                                  0,
+                                  0,
+                                  sec['NBT'],
+                                  sec['INT'],
+                                  sec['TINT'],
+                                  sec['W'])
+    return "\n".join(sandy.write_eol(lines, sec["MAT"], 7, sec["MT"]))
+
+
+def _write_inelastic_scattering(sec):
+    """
+    Given the content of MF7 for inelastic scattering as nested dictionaries,
+    write it to string.
+
+    Parameters
+    ----------
+    sec : 'dict'
+        Multiline string reproducing the content of a ENDF-6 section.
+
+    Returns
+    -------
+    `str`
+        Multiline string reproducing the content of a ENDF-6 section.
+
+    Examples
+    --------
+    Incoherent inelastic scattering:
+    >>> tls = sandy.get_endf6_file("endfb_80", 'tsl', 26)
+    >>> sec = sandy.sections.mf7._read_incoherent_inelastic(tls, 26, 4)
+    >>> assert len(_write_inelastic_scattering(sec)) == len(tls.data[(26, 7, 4)])
```

Comment:
Can you check why this example cannot work for mat = 1 or 11, 40, 47, 49, 51?

Reply:
I will check, thanks